src/components/PDFView/PDFPasswordForm.js

```diff
@@ -0,0 +1,135 @@
+            password: '',
+            validationErrorText: '',
+            isEditingInProgress: true,
+        };
+        this.submitPassword = this.submitPassword.bind(this);
+        this.validateOnBlur = this.validateOnBlur.bind(this);
+        this.updatePassword = this.updatePassword.bind(this);
+    }
+
+    submitPassword() {
+        this.validateOnBlur();
+        if (!_.isEmpty(this.state.password)) {
+            this.setState({isEditingInProgress: false});
+            this.props.onSubmit(this.state.password);
+        }
+    }
+
+    updatePassword(password) {
+        if (!_.isEmpty(password)) {
+            this.setState({validationErrorText: ''});
+        }
+        this.setState({password, isEditingInProgress: true});
+    }
+
+    validateOnBlur() {
+        if (!_.isEmpty(this.state.password)) {
+            return;
+        }
+        this.setState({
+            validationErrorText: this.props.translate('attachmentView.passwordRequired'),
+        });
+    }
+
+    render() {
+        // Use container styles appropriate for screen size.
+        const containerStyles = this.props.isSmallScreenWidth
+            ? styles.pdfPasswordForm.narrowScreen
+            : styles.pdfPasswordForm.wideScreen;
+
+        const buttonStyles = [
+            styles.pdfPasswordForm.confirmButton,
+            {paddingBottom: this.props.insets.bottom},
+        ];
+
+        return (
+            <View style={containerStyles}>
+                <Text style={styles.mb4}>
+                    {this.props.translate('attachmentView.pdfPasswordFormLabel')}
+                </Text>
+                <TextInput
+                    label={this.props.translate('common.password')}
+                    autoCompleteType="off"
+                    autoCorrect={false}
+                    textContentType="password"
+                    onInputChange={this.updatePassword}
+                    returnKeyType="done"
+                    onSubmitEditing={this.submitPassword}
+                    errorText={this.state.validationErrorText}
+                    onBlur={this.validateOnBlur}
+                    autoFocus={this.props.shouldAutofocusPasswordField}
+                    secureTextEntry
+                />
+                {!this.state.isEditingInProgress && this.props.isPasswordInvalid && (
+                    <View style={[styles.flexRow, styles.alignItemsCenter, styles.mt3]}>
+                        <Icon src={Expensicons.Exclamation} fill={colors.red} />
+                        <View style={[styles.flexRow, styles.ml2, styles.flexWrap, styles.flex1]}>
+                            <Text style={styles.mutedTextLabel}>
+                                {this.props.translate('attachmentView.passwordIncorrect')}
+                            </Text>
+                        </View>
+                    </View>
+                )}
+                {this.props.isSmallScreenWidth && (
+                    <View style={{flexGrow: 1}} />
```

Comment:
Also, we don't allow inline styles in our codebase.  Use generic styles from styles.js.